oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/query/IMetadataQueryDAO.java

```diff
@@ -64,7 +64,17 @@ List<ServiceInstance> listInstances(final long startTimestamp, final long endTim
      * @param agentId the agent id which reports the process.
      * @return list of processes matching the given conditions.
      */
-    List<Process> listProcesses(final String serviceId, final String instanceId, final String agentId) throws IOException;
+    List<Process> listProcesses(final String serviceId, final String instanceId, final String agentId,
+                                final long lastPingStartTimeBucket, final long lastPingEndTimeBucket) throws IOException;
+
+    /**
+     * get the count of processes
+     * @param serviceId the service of the processes.
+     * @param instanceId the service instance of the process.
+     * @param agentId the agent id which reports the process.
+     * @return the size of processes
+     */
+    long getProcessesCount(final String serviceId, final String instanceId, final String agentId) throws IOException;
```

Comment:
Whether these two should be merged? I am feeling these are the same thing?

Reply:
I also thought about this. the `listProcesses` just list all the processes, including the metadata of the processes. 
But `getProcesses` is a quick way to query the count of simple metadata, doesn't need the metadata from the storage, and it could speed up the query time.

Reply:
OK, from a performance perspective, this makes sense with two methods. But, to get the number of processes, the time range is important. Then

Reply:
Updated.